File: 1122 - Relative Sort Array.py
class Solution:
    def relativeSortArray(self, arr1: list[int], arr2: list[int]) -> list[int]:
        res = []
        remain = []
        added = [False] * len(arr1)
        s = set(arr2)
        for i in range(len(arr2)):
            for j, n in enumerate(arr1):
                if added[j]:
                    continue
                if n == arr2[i]:
                    res.append(n)
                    added[j] = True
                elif n not in s:
                    remain.append(n)
                    added[j] = True
        remain.sort()
        res += remain
        return res

File: 1122 - Relative Sort Array.cpp
class Solution {
public:
    vector<int> relativeSortArray(vector<int>& arr1, vector<int>& arr2) {
        int f[1001]{0};
        vector<int> res(arr1.size());
        for (int n : arr1) {
            f[n]++;
        }
        int idx = 0;
        for (int n : arr2) {
            for (int i = 0; i < f[n]; i++) {
                res[idx++] = n;
            }
            f[n] = 0;
        }
        for (int i = 0; i < 1001; i++)
        {
            for (int j = 0; j < f[i]; j++) {
                res[idx++] = i;
            }
        }
        return res;
        

    }
};